spark_ec2.py

```diff
@@ -144,15 +144,36 @@ def setup_external_libs(libs):
             tar.close()
             os.remove(tgz_file_path)
             print(" - Finished downloading {lib}.".format(lib=lib["name"]))
-        sys.path.insert(1, lib_dir)
+
+        if lib["add-path"]:
+            lib["add-path"](lib_dir)
+        else:
+            sys.path.insert(1, lib_dir)
+
+
+def add_pyaml_path(lib_dir):
+    lib_py2 = os.path.join(lib_dir, "lib")
+    if os.path.exists(lib_py2) and sys.version < "3":
+        sys.path.insert(1, lib_py2)
+
+    lib_py3 = os.path.join(lib_dir, "lib3")
+    if os.path.exists(lib_py3) and sys.version >= "3":
+        sys.path.insert(1, lib_py3)
 
 
 # Only PyPI libraries are supported.
 external_libs = [
     {
         "name": "boto",
         "version": "2.34.0",
-        "md5": "5556223d2d0cc4d06dd4829e671dcecd"
+        "md5": "5556223d2d0cc4d06dd4829e671dcecd",
+        "add-path": None
```

Comment:
The purpose `add-path` is to precise a custom function to execute, because some libraries have to be added to the system path differently than just doing sys.path.insert(1, libs/LIBRARY_NAME).